auth-manager/src/main/java/com/qpay/authmanager/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.qpay.authmanager.service.impl;
+
+import com.qpay.authmanager.mapper.UserMapper;
+import com.qpay.authmanager.model.dto.UserModification;
+import com.qpay.authmanager.model.entity.UserEntity;
+import com.qpay.authmanager.repository.UserRepository;
+import com.qpay.authmanager.service.UserService;
+import com.qpay.authmanager.service.exception.UserAlreadyExistsException;
+import com.qpay.authmanager.service.exception.UserNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+
+    private final UserMapper userMapper;
+
+    private final PasswordEncoder passwordEncoder;
+
+    public UserEntity addUser(final UserModification userModification) {
+        existUser(userModification.email());
```

Comment:
bad name. It says that you check that user exists. But in reality if they exist you throw exception.
```suggestion
        checkUserDoesNotExist(userModification.email());
```